src/test/java/qna/domain/AnswerRepositoryTest.java

```diff
@@ -2,48 +2,77 @@
             () -> assertThat(actual.isDeleted()).isFalse(),
-            () -> assertThat(actual.getQuestionId()).isEqualTo(A1.getQuestionId()),
-            () -> assertThat(actual.getWriterId()).isEqualTo(A1.getWriterId())
+            () -> assertThat(actual.getQuestion()).isEqualTo(answer.getQuestion()),
+            () -> assertThat(actual.getContents()).isEqualTo(answer.getContents()),
+            () -> assertThat(actual.getWriter()).isEqualTo(answer.getWriter())
         );
-        assertThat(answerRepository.findByIdAndDeletedFalse(actual.getId())).contains(actual);
+        assertThat(answerRepository.findByIdAndDeletedFalse(actual.getId())).contains(answer);
     }
 
     @Test
     void 답변_삭제여부_변경() {
-        Answer actual = answerRepository.save(A1);
-        assertThat(answerRepository.findByIdAndDeletedFalse(actual.getId())).isNotEmpty();
-        actual.setDeleted(true);
-        assertThat(answerRepository.findByIdAndDeletedFalse(actual.getId())).isEmpty();
+        User questionWriter = userRepository.save(newUser("1"));
+        Question question = questionRepository.save(newQuestion("1"));
+        question.writeBy(questionWriter);
+        User answerWriter = userRepository.save(newUser("2"));
+        Answer answer = new Answer(answerWriter, question, "answerContents");
+        Answer savedAnswer = answerRepository.save(answer);
+
+        Answer findAnswer = answerRepository.findByIdAndDeletedFalse(savedAnswer.getId())
+            .orElseThrow(RuntimeException::new);
+        assertThat(findAnswer).isEqualTo(savedAnswer);
+        findAnswer.changeDeleted(true);
+        assertThat(answerRepository.findByIdAndDeletedFalse(findAnswer.getId())).isEmpty();
     }
 
     @Test
     void 삭제되지않은_질문에대한_답변들_찾기() {
-        Answer savedA1 = answerRepository.save(A1);
-        Answer savedA2 = answerRepository.save(A2);
-        List<Answer> answers = answerRepository.findByQuestionIdAndDeletedFalse(A1.getQuestionId());
-        assertThat(answers).hasSize(2);
-        assertThat(answers).containsExactly(savedA1, savedA2);
+        User questionWriter = userRepository.save(newUser("1"));
+        Question question = newQuestion("1");
+        Question savedQuestion = questionRepository.save(question);
+        savedQuestion.writeBy(questionWriter);
+
+        User answerWriter = userRepository.save(newUser("2"));
+        Answer savedAnswer1 = answerRepository.save(new Answer(answerWriter, savedQuestion, "answerContents1"));
+        Answer savedAnswer2 = answerRepository.save(new Answer(answerWriter, savedQuestion, "answerContents2"));
+
+        Question findQuestion = questionRepository.findByIdAndDeletedFalse(savedQuestion.getId())
+            .orElseThrow(RuntimeException::new);
+
+        List<Answer> findAnswers = findQuestion.getAnswers();
+        assertThat(findAnswers).hasSize(2);
+        assertThat(findAnswers).containsExactly(savedAnswer1, savedAnswer2);
```

Comment:
알고 계실수도 있지만 
assertj 를 이용하시면 다음과 같이 `Optional`과 리스트를 한번에 테스트할 수 있으니 참고만해주세요!!

```suggestion
        assertThat(questionRepository.findByIdAndDeletedFalse(savedQuestion.getId()))
                .get()
                .extracting(Question::getAnswers, InstanceOfAssertFactories.LIST)
                .hasSize(2)
                .containsExactly(savedAnswer1, savedAnswer2);
```

Reply:
이렇게 메소드 체이닝으로 한번에 확인할 수도 있군요!!
잘 활용하겠습니다. 감사합니다 👍 👍